src/Common/src/Interop/Linux/procfs/Interop.ProcFsStat.cs

```diff
@@ -0,0 +1,321 @@
+            internal long priority;
+            internal long nice;
+            internal long num_threads;
+            internal long itrealvalue;
+            internal ulong starttime;
+            internal ulong vsize;
+            internal long rss;
+            internal ulong rsslim;
+            internal ulong startcode;
+            internal ulong endcode;
+            internal ulong startstack;
+            internal ulong kstkesp;
+            internal ulong kstkeip;
+            internal ulong signal;
+            internal ulong blocked;
+            internal ulong sigignore;
+            internal ulong sigcatch;
+            internal ulong wchan;
+            internal ulong nswap;
+            internal ulong cnswap;
+            internal int exit_signal;
+            internal int processor;
+            internal uint rt_priority;
+            internal uint policy;
+            internal ulong delayacct_blkio_ticks;
+            internal ulong guest_time;
+            internal long cguest_time;
+        }
+
+        private static string GetStatFilePathForProcess(int pid)
+        {
+            return RootPath + pid.ToString(CultureInfo.InvariantCulture) + "/stat";
+        }
+
+        internal static string GetTaskDirectoryPathForProcess(int pid)
+        {
+            return RootPath + pid.ToString(CultureInfo.InvariantCulture) + "/task/";
+        }
+
+        private static string GetStatFilePathForThread(int pid, int tid)
+        {
+            // Perf note: Calling GetTaskDirectoryPathForProcess will allocate a string,
+            // which we then use in another Concat call to produce another string.  The straightforward alternative,
+            // though, since we have five input strings, is to use the string.Concat overload that takes a params array. 
+            // This results in allocating not only the params array but also a defensive copy inside of Concat,
+            // which means allocating two five-element arrays.  This two-string approach will result not only in fewer 
+            // allocations, but also typically in less memory allocated, and it's a bit more maintainable.
+            return GetTaskDirectoryPathForProcess(pid) + tid.ToString(CultureInfo.InvariantCulture) + "/stat";
+        }
+
+        internal static ParsedStat ReadStatFile(int pid)
+        {
+            string stat = File.ReadAllText(GetStatFilePathForProcess(pid));
+            return ParseStatFile(stat);
+        }
+
+        internal static ParsedStat ReadStatFile(int pid, int tid)
+        {
+            string stat = File.ReadAllText(GetStatFilePathForThread(pid, tid));
+            return ParseStatFile(stat);
+        }
+
+        private static ParsedStat ParseStatFile(string statFileContents)
+        {
+            ParsedStat results = default(ParsedStat);
+
+            int index = 0;
+            int i = 0;
+            for (; i < ExpectedStatFileParts && index < statFileContents.Length; i++)
```

Comment:
> Would it be possible to structure the code in the following way?

Excellent suggestion; I've done that and it not only helps clean up the code, it makes it easier to skip over creating strings for fields we're not actually using anywhere.

> In Debug builds, we would assert, but in release builds we'll just return a ParsedStat structure that is incomplete. Is that what we actually want?

I went back and forth on this.  Since you obviously lean towards making it exceptional, that's what I've now done.